src/main/java/com/pinterest/secor/uploader/S3UploadManager.java

```diff
@@ -117,11 +119,23 @@ public void refresh() {}
 
     public Handle<?> upload(LogFilePath localPath) throws Exception {
         String s3Bucket = mConfig.getS3Bucket();
-        String s3Key = localPath.withPrefix(mConfig.getS3Path()).getLogFilePath();
+        Date s3MigrationDate = null;
+        String s3Path;
+        try {
+            s3MigrationDate = new SimpleDateFormat("yyyy-MM-dd").parse(mConfig.getS3MigrationDate());
```

Comment:
S3MigrateDate and S3MigratePath should be evaluated in UploadManager's construction time.